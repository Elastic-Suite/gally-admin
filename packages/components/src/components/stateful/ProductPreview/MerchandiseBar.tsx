import React from 'react'
import { Grid, Paper, styled } from '@mui/material'
import { useTranslation } from 'next-i18next'
import {
  CustomNoTopProduct,
  SearchResult,
  SearchTitle,
} from '../Merchandize/SearchBar/Search.styled'

const PaperStyled = styled(Paper)(() => ({
  display: 'flex',
}))

const GridStyled = styled(Grid)(({ theme }) => ({
  padding: theme.spacing(2),
}))

interface IProps {
  nbResults: number
  nbTopProducts: number
}

export default function MerchandiseBar(props: IProps): JSX.Element {
  const { nbResults, nbTopProducts } = props
  const { t } = useTranslation('categories')

  const value = {
    value: nbResults + nbTopProducts,
    result: t('searchBar.results', { count: nbResults }),
  }

  const result = t('searchBarResult', { value })

  const valuePinned = {
    value: nbTopProducts,
    result: t('pinned.results', { count: nbTopProducts }),
  }
  const resultPinned = t('searchBarResult', { value: valuePinned })

  const hasProduct = Boolean(nbTopProducts + nbResults)

  return (
    <PaperStyled variant="outlined">
      <GridStyled container direction="column">
        <SearchTitle>{t(nbTopProducts + nbResults == 1 ? 'merchandise.singular.title' : 'merchandise.plural.title')}</SearchTitle>
        <SearchResult>
          {!hasProduct ? (
            <CustomNoTopProduct>{t('noProductSearch')}</CustomNoTopProduct>
          ) : (
            <>
              {result} {` (${resultPinned})`}
            </>
          )}
        </SearchResult>
      </GridStyled>
    </PaperStyled>
  )
}
