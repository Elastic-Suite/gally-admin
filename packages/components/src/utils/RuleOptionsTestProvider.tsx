import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
