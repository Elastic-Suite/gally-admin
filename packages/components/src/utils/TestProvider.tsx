import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
