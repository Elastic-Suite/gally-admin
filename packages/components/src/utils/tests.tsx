import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
