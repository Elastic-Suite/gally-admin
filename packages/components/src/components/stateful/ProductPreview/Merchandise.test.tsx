import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import MerchandiseBar from './MerchandiseBar'

describe('MerchandiseBar match snapshot', () => {
  it('MerchandiseBar', () => {
    const { container } = renderWithProviders(
      <MerchandiseBar nbResults={25} nbTopProducts={2} />
    )
    expect(container).toMatchSnapshot()
  })
})
