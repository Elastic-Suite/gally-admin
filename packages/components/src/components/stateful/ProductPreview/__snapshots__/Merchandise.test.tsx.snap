// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchandiseBar match snapshot MerchandiseBar 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-1m4bt7v-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-jddpo1-MuiGrid-root"
    >
      <div
        class="MuiGrid-root css-1pan2h9-MuiGrid-root"
      >
        searchBar.title
      </div>
      <div
        class="MuiGrid-root css-1bhfatf-MuiGrid-root"
      >
        searchBarResult
         
         (searchBarResult)
      </div>
    </div>
  </div>
</div>
`;
